index.js

```diff
@@ -7,26 +7,33 @@ import normalizePackageData from 'normalize-package-data';
 
 const toPath = urlOrPath => urlOrPath instanceof URL ? fileURLToPath(urlOrPath) : urlOrPath;
 
-export async function readPackage({cwd, normalize = true} = {}) {
-	cwd = toPath(cwd) || process.cwd();
-	const filePath = path.resolve(cwd, 'package.json');
-	const json = parseJson(await fsPromises.readFile(filePath, 'utf8'));
+const getPackagePath = cwd => {
+	const packageDir = toPath(cwd) || process.cwd();
+	return path.resolve(packageDir, 'package.json');
+};
+
+const _readPackage = (file, normalize) => {
+	const json = typeof file === 'string'
+		? parseJson(file)
+		: file; // TODO: ensure `file` is an object here
```

Comment:
Is it enough to check for `typeof file === 'object'` here?

Reply:
`file !== null && typeof file === 'object'`

Reply:
Should probably throw a TypeError if it's neither a string or object.